jme3-desktop/src/main/java/com/jme3/system/NativeLibraryLoader.java

```diff
@@ -571,89 +478,80 @@ public static void loadNativeLibrary(String name, boolean isRequired) {
+                    logger.log(Level.FINE, "Not copying library {0}. Latest already extracted.",
+                            loadedAsFileName);
                 }
             }
 
-            out = new FileOutputStream(targetFile);
-            int len;
-            while ((len = in.read(buf)) > 0) {
-                out.write(buf, 0, len);
-            }
-            
-            in.close();
-            in = null;
-            out.close();
-            out = null;
+            library.getLoadFunction().accept(targetFile.getAbsolutePath());
 
-            // NOTE: On OSes that support "Date Created" property, 
-            // this will cause the last modified date to be lower than
-            // date created which makes no sense
-            targetFile.setLastModified(conn.getLastModified());
+            if (logger.isLoggable(Level.FINE)) {
+                logger.log(Level.FINE, "Loaded native library {0}.", library.getName());
+            }
         } catch (IOException ex) {
-            if (ex.getMessage().contains("used by another process")) {
+            /*if (ex.getMessage().contains("used by another process")) {
                 return;
-            } else {
-                throw new UncheckedIOException("Failed to extract native "
-                        + "library to: " + targetFile, ex);
-            }
-        } finally {
-            // XXX: HACK. Vary loading method based on library name.
-            // lwjgl and jinput handle loading by themselves.
-            if (name.equals("lwjgl") || name.equals("lwjgl3")) {
-                System.setProperty("org.lwjgl.librarypath", 
-                                   extractionDirectory.getAbsolutePath());
-            } else if (name.equals("jinput")) {
-                System.setProperty("net.java.games.input.librarypath", 
-                                   extractionDirectory.getAbsolutePath());
-            } else {
-                // all other libraries (openal, bulletjme, custom)
-                // will load directly in here.
-                System.load(targetFile.getAbsolutePath());
-            }
-            
-            if(in != null){
-                try { in.close(); } catch (IOException ex) { }
-            }
-            if(out != null){
-                try { out.close(); } catch (IOException ex) { }
-            }
+            }*/
+
+            throw new UncheckedIOException("Failed to extract native library to: "
+                    + targetFile, ex);
         }
+    }
+
+    /**
+     * Checks if library extraction is required by comparing source and target
+     * last modified date. Returns true if target file does not exist.
+     *
+     * @param conn the source file
+     * @param targetFile the target file
+     * @return false if target file exist and the difference in last modified date is
+     *          less than 1 second, true otherwise
+     */
+    private static boolean isLibraryExtractionRequired(URLConnection conn, File targetFile) {

```

Comment:
Nice work, btw when a function has more than one return, i guess ordering those returns ascendingly by complexity will give the reader a better idea of  the function job, so for example: 
```java
private static boolean isLibraryExtractionRequired(URLConnection conn, File targetFile) {
     // extract anyway if the file doesn't exist 
     if (!targetFile.exists()) {
         return true;  
     }
    // if the file exists then:
    // compare last modified date of this file to file in jar
    // Allow ~1 second range for OSes that only support low precision
    return Math.abs(conn.getLastModified() - targetFile.lastModified()) >= 1000;
}
```

Reply:
If you are using intellij, it may show you an equivalent shortcut to my proposed change, like this:
```java
private static boolean isLibraryExtractionRequired(URLConnection conn, File targetFile) {
     // extract anyway if the file doesn't exist 
     // if the file exists then:
    // compare last modified date of this file to file in jar
    // Allow ~1 second range for OSes that only support low precision
    return !targetFile.exists() || Math.abs(conn.getLastModified() - targetFile.lastModified()) >= 1000;
}
```
You may choose to keep it like that too, it's the same, but i don't prefer this syntax...

Reply:
Done, 

Thanks for the feedback :)